### ISSUE TITLE
Interruption during playback

### ISSUE THREAD

**Issue by dahumph** [balbuze/volumio-plugins#246]
Hi, I'm running latest brutefir plugin on latest volumio version on RPI3. Everything works fine, except that playback is interrupted approx. once a minute for about 1 second. Investigation shows that at time of interruptions brutefir.real processes are closing and new brutefir.real PIDs are started. Probably the process crashes and restarts, continues to work for a few seconds or minutes and so on. Tried it with *.dbl and *.txt filters, both resulting in same issue.

Any idea how to improve on that? Wondering if anybody else has that issue?

**Comment by balbuze** [comment]
Hi, do you use a usb Dac? What version of the plugin?

**Comment by dahumph** [comment]
I'm not using any usb devices, only a Mytec DAC is connected via a hifiberry digi+ SPDIF out. Plugin version is from 3rd November. But I had that issue with older verions, even with brutefir2 years ago. Is there a verbose mode or a logfile so I can check why the brutefir.real processes are closing after a short time?

**Comment by dahumph** [comment]
Anybody? Wondering if I'm the only one having those drops while using almost a standard setup with RPI3 and a hifiberry soundcard

**Comment by balbuze** [comment]
You can check if error occurs while playing using 'sudo journactl -f'

**Comment by dahumph** [comment]
Ok thanks. Basically it is a broken pipe problem. The relevant output is as following:

Nov 11 14:35:26 volumio volumio[6285]: ALSA I/O: Could not read audio: Broken pipe.
Nov 11 14:35:26 volumio volumio[6285]: I/O module failed to read (probably) due to buffer underflow on output, aborting.
Nov 11 14:35:26 volumio systemd[1]: brutefir.service: main process exited, code=exited, status=5/NOTINSSTALLED
Nov 11 14:35:26 volumio systemd[1]: Unit brutefir.service entered failed state.
Nov 11 14:35:27 volumio systemd[1]: brutefir.service holdoff time over, scheduling restart.
Nov 11 14:35:27 volumio systemd[1]: Stopping brutefir Daemon...
Nov 11 14:35:27 volumio systemd[1]: Starting brutefir Daemon...
Nov 11 14:35:27 volumio systemd[1]: Started brutefir Daemon.
Nov 11 14:35:27 volumio volumio[6990]: BruteFIR v1.0m (November 2013)                                (c) Anders Torger
Nov 11 14:35:27 volumio volumio[6990]: Internal resolution is 64 bit floating point.
Nov 11 14:35:27 volumio volumio[6990]: Creating 4 FFTW plans of size 4096...finished.
Nov 11 14:35:27 volumio volumio[6990]: Loading 16 coefficient sets...finished.
Nov 11 14:35:27 volumio volumio[6990]: Realtime priorities are min = 2, usermax = 1, mid = 3 and max = 4.
Nov 11 14:35:27 volumio volumio[6990]: Warning: no support for clock cycle counter on this platform.
Nov 11 14:35:27 volumio volumio[6990]: Timers for benchmarking may be unreliable.
Nov 11 14:35:27 volumio volumio[6990]: Realtime priority 1 set for cli process (pid 6996)
Nov 11 14:35:27 volumio volumio[6990]: Realtime priority 3 set for input process (pid 6991)
Nov 11 14:35:27 volumio volumio[6990]: Realtime priority 4 set for filter process (pid 6995)
Nov 11 14:35:27 volumio volumio[6990]: Realtime priority 4 set for filter process (pid 6994)
Nov 11 14:35:27 volumio volumio[6990]: Realtime priority 3 set for output process (pid 6997)
Nov 11 14:35:27 volumio volumio[6990]: Fixed I/O-delay is 4096 samples
Nov 11 14:35:27 volumio volumio[6990]: Audio processing starts now


This error occures peridodically approximately every 1 second. During playback and strangely also during idle. It brakes even if no filters are set. 

Any idea where I can start to fix it?

**Comment by balbuze** [comment]
I use the plugin with a RPI3 + allo boss DAC and it works perfecly. So I think we can exclude a problem from the RPI... Do you use something else (plugin or other program) on the device?

**Comment by dahumph** [comment]
There is no other service running, clean volumio installation and only the DRC plugin installed, so I suspected an issue with the Hifiberry digi card and unmounted it. Connected to a DAC via USB  =>no more errors. 

So at this point we can conclude it's most probably a compatibility issue with the hifiberry digi+ board.

Any suggestion where to start fixing is highly welcome 

:-(

**Comment by ogroot** [comment]
Did you solve this problem? 

I encounter a similar issue.  Running 9 months with Volumio/DRC plugin, without problems.
After upgrading to 2.699 and DRC 1.4.2 I also have frequent interruptions. 

If I lower the bitrate of DRC it gets better, but interruptions still occur. 
I'm running on a Pi2B could it be that newer versions are compiled differently?

**Comment by balbuze** [comment]
Do you remember your last working version of the plugin?or approximately the date? Some other users reports the same problem, but I don't met it until know. So it is not easy to understand what happens. When using 'sudo journalctl -f' while playing, what happens when it stop? Can you see some errors?

**Comment by ogroot** [comment]
Hi Balbuze, Thanks for your response.....

I added a new issue in Jira (#254) with the 'sudo journalctl -f'  output, because I thought the issue was slightly different then this thread.  I also did some further testing with versions
Results are:

Volumio 2.598 + DRC 1.1.5 --> 44.1khz = OK, 48khz = OK, 96khz = OK
Volumio 2.598 + DRC 1.4.2 --> 44.1khz = occasional drop, 48khz = NOT OK, 96khz = NOT OK
Volumio 2.699 + DRC 1.4.2 --> 44.1khz = occasional drop, 48khz --> NOT OK, 96khz = NOT OK

**Comment by dahumph** [comment]
> Did you solve this problem with the Hifiberry digi?

No, not at all. But I can say that for me this issue is persisting for 2 years now, I tried it from time to time with the latest versions of Volumio and the plugin. For my setup with RPI3 and the HifiBerry digi+ board it never worked, regardless which Volumio version.
